engine/swarm/secrets.md

```diff
@@ -0,0 +1,837 @@
+    can't be used to sign certificates.
+
+    ```none
+    [server]
+    authorityKeyIdentifier=keyid,issuer
+    basicConstraints = critical,CA:FALSE
+    extendedKeyUsage=serverAuth
+    keyUsage = critical, digitalSignature, keyEncipherment
+    subjectAltName = DNS:localhost, IP:127.0.0.1
+    subjectKeyIdentifier=hash
+    ```
+
+8.  Sign the site certificate.
+
+    ```bash
+    $ openssl x509 -req -days 750 -in "site.csr" -sha256 \
+        -CA "root-ca.crt" -CAkey "root-ca.key"  -CAcreateserial \
+        -out "site.crt" -extfile "site.cnf" -extensions server
+    ```
+
+9.  The `site.csr` and `site.cnf` files are not needed by the Nginx service, but
+    you will need them if you want to generate a new site certificate. Protect
+    the `root-ca.key` file.
+
+#### Configure the Nginx container
+
+1.  Produce a very basic Nginx configuration that serves static files over HTTPS.
+    The TLS certificate and key will be stored as Docker secrets so that they
+    can be rotated easily.
+
+    In the current directory, create a new file called `site.conf` with the
+    following contents:
+
+    ```none
+    server {
+        listen                443 ssl;
+        server_name           localhost;
+        ssl_certificate       /run/secrets/site.crt;
+        ssl_certificate_key   /run/secrets/site.key;
+
+        location / {
+            root   /usr/share/nginx/html;
+            index  index.html index.htm;
+        }
+    }
+    ```
+
+2.  Create three secrets, representing the key, the certificate, and the
+    `site.conf`. You can store any file as a secret as long as it is smaller
+    than 500 KB. This allows you to decouple the key, certificate, and
+    configuration from the services that will use them.
+
+    ```bash
+    $ cat site.key | docker secret create site.key
+
+    $ cat site.crt | docker secret create site.crt
+
+    $ cat site.conf | docker create site.conf
+    ```
+
+    ```bash
+    $ docker secret ls
+
+    ID                          NAME                  CREATED             UPDATED
+    2hvoi9mnnaof7olr3z5g3g7fp   site.key       58 seconds ago      58 seconds ago
+    aya1dh363719pkiuoldpter4b   site.crt       24 seconds ago      24 seconds ago
+    zoa5df26f7vpcoz42qf2csth8   site.conf      11 seconds ago      11 seconds ago
+    ```
+
+4.  Create a service that runs Nginx and has access to the three secrets. The
+    last part of the `docker service create` command creates a symbolic link
+    from the location of the `site.conf` secret to `/etc/nginx.conf.d/`, where
+    Nginx looks for extra configuration files. This step happens before Nginx
+    actually starts, so you don't need to rebuild your image if you change the
+    Nginx configuration.
+
+    > **Note**: Normally you would create a Dockerfile which copies the `site.conf`
+    > into place, build the image, and run a container using your custom image.
+    > This example does it all in one step.
```

Comment:
Maybe something like "This example does not require a custom image; it copies the `site.conf` into place and runs the container all in one step"?